pkg/networkservice/common/refresh/client.go

```diff
@@ -59,25 +59,37 @@ func (t *refreshClient) Request(ctx context.Context, request *networkservice.Net
 	req := request.Clone()
 	// Set its connection to the returned connection we received
 	req.Connection = rv
-	// Setup the timer with the req containing the returned connection
 
 	expire, err := t.getExpireDuration(request)
 	if err != nil {
 		return nil, errors.Wrapf(err, "Error creating timer from Request.Connection.Path.PathSegment[%d].ExpireTime", request.GetConnection().GetPath().GetIndex())
 	}
 	t.executor.AsyncExec(func() {
-		if refreshCtx := refreshContext(ctx); refreshCtx != nil && refreshCtx.Err() != nil {
-			return
-		}
-		if timer, ok := t.connectionTimers[req.GetConnection().GetId()]; ok {
-			timer.Stop()
-		}
-		if canceller, ok := t.refreshCancellers[req.GetConnection().GetId()]; ok {
-			canceller()
+		// check if it is refresh request
+		if refreshCtx := refreshContext(ctx); refreshCtx != nil {
+			// refresh was canceled
+			if refreshCtx.Err() != nil {
+				return
+			}
+			// we reuse non-canceled refresh context for the next refresh request
+			timer := t.createTimer(ctx, req, expire, opts...)
+			t.connectionTimers[req.GetConnection().GetId()] = timer
```

Comment:
[minor] `req.GetConnection().GetId()` reused many times, you can create local variable for it. 
```suggestion
			id := req.GetConnection().GetId()
			t.connectionTimers[id] = timer
```